### ISSUE TITLE
Displayed date format and its inconsistency

### ISSUE THREAD

**Issue by pierreporte** [OtterBrowser/otter-browser#1160]
Screenshot of the history first:

![capture du 2016-07-01 18-47-15](https://cloud.githubusercontent.com/assets/2029853/16528802/59c60738-3fbf-11e6-8eea-2492ccf0c526.png)

When you open the completion menu directly without entering anything in the URL entry field (number 1 on the screenshot), the history is displayed. The dates associated with entries is displayed using the default format usually found in databases and is not really meant to be read. By comparison, the classical history displays the date and hour in a more natural way. The history in the completion menu should have a date and hour formatted better. This leads me to the following.

The history manager uses folders to sort entries by access time : today (point 2), earlier this week (point 3) and earlier this year (point 3). (I think that it lacks a folder for URLs accessed earlier this month.) For today’s URLs, we still have the whole date. It would be clearer just to display the hour. For this week’s URLs, we don’t need the month and day number: the day name and the hour are sufficient. For the rest, it is OK yet I would personally prefer having long dates (e.g. 1st of July) instead of short dates but it could be a setting. The new date formats should be applied in the completion menu as well.

The non-human readable date format is also used in several parts of the browser like the properties of bookmarks.

To sum up:
- For URLs accessed today, display just the hour, and in the completion menu “today at 13:37”,
- For URLs accessed earlier this week, display something like “monday at 13:37”,
- For URLs accessed earlier this year, display something like “1st of July at 13:37” (or 01/07/2016 at 13:37 if short dates are chosen),
- For URLs accessed last year, just add the year in long date format,
- Display this kind of dates in the whole browser (except maybe in about:config).

**Comment by Emdek** [comment]
@pierreporte, special cases for dates are a bit problematic, since it requires scheduled redrawing for cases where such page is opened long enough to invalidate result.
It is still doable, but I would make it only optional, perhaps enabled by default but with option to disable it.
